tempodb/encoding/vparquet/create.go

```diff
@@ -215,3 +227,89 @@ func (b *streamingBlock) Complete() (int, error) {
 
 	return n, writeBlockMeta(b.ctx, b.to, b.meta, b.bloom)
 }
+
+// estimateTraceSize attempts to estimate the size of trace in bytes. This is used to make choose
+// when to cut a row group during block creation.
+// TODO: This function regularly estimates lower values then estimateProtoSize() and the size
```

Comment:
Yeah, throwing in an extra byte per field does bring it a lot closer. I suppose that consistency is as important as accuracy here. As long as it's consistent it will respond in expected ways to adjusting the row sizes.

Currently having both of these methods is a little tedious though. Hopefully we will find ways to get the compaction cycle to use GenericReader/Writer effectively and we can just have one of these methods.